CSharp/DevVmPowershell/Helpers/Implementations/ProcessingHelper.cs

```diff
@@ -285,61 +207,21 @@ public async Task<bool> CreateWorkerManagerServerAsync()
-			{
-				Field = Constants.Processing.NameField,
-				Operator = Relativity.Services.TextConditionEnum.EqualTo,
-				Value = Constants.Processing.ResourceServerName
-			};
-
-			Relativity.Services.Query queryServer = new Relativity.Services.Query()
+			HttpClient httpClient = RestHelper.GetHttpClient(ConnectionHelper.RelativityInstanceName, ConnectionHelper.RelativityAdminUserName, ConnectionHelper.RelativityAdminPassword);
+			// Check if Worker Manager Server is already exists
+			if (await GetWorkerManagerServerArtifactIdAsync(httpClient) == -1)
 			{
-				Condition = conditionServer.ToQueryString()
-			};
-
-			// The Worker Manager Server to be created
-			WorkerManagerServer server = new WorkerManagerServer()
-			{
-				Name = Constants.Processing.ResourceServerName,
-				IsDefault = true,
-				InventoryPriority = 100,
-				DiscoveryPriority = 100,
-				PublishPriority = 100,
-				ImageOnTheFlyPriority = 1,
-				MassImagingPriority = 100,
-				SingleSaveAsPDFPriority = 100,
-				MassSaveAsPDFPriority = 100,
-				URL = Constants.Processing.ResourceServerUrl
-			};
-
-			using (IWorkerServerManager workerServerManager = ServiceFactory.CreateProxy<IWorkerServerManager>())
+				// Create Worker Manager Server
+				await CallCreateWorkerManagerServerAsync(httpClient);
+				Console.WriteLine($"{nameof(CreateWorkerManagerServerAsync)} - Successfully created Worker Manager Server ({Constants.Processing.ResourceServerName})");
+				wasWorkerManagerServerCreated = true;
+			}
+			else
 			{
-				WorkerServerQueryResultSet queryResult = await workerServerManager.QueryAsync(queryServer);
-
-				if (queryResult.Success && queryResult.TotalCount == 0)
-				{
-					int workerServerArtifactId = await workerServerManager.CreateSingleAsync(server);
-					Console.WriteLine($"{nameof(CreateWorkerManagerServerAsync)} - Worker Server Artifact ID: {workerServerArtifactId}");
-
-					queryResult = await workerServerManager.QueryAsync(queryServer);
-
-					if (queryResult.Success && queryResult.TotalCount > 0)
-					{
-						wasWorkerManagerServerCreated = true;
-						Console.WriteLine($"{nameof(CreateWorkerManagerServerAsync)} - Successfully created Worker Manager Server ({Constants.Processing.ResourceServerName})");
-					}
-				}
-				else if (queryResult.Success && queryResult.TotalCount > 0)
-				{
-					Console.WriteLine($"{nameof(CreateWorkerManagerServerAsync)} - Failed to create Worker Manager Server ({Constants.Processing.ResourceServerName}) as it already exists");
-					wasWorkerManagerServerCreated = true;
-				}
-				else
-				{
-					Console.WriteLine($"{nameof(CreateWorkerManagerServerAsync)} - Failed to create and check for Worker Manager Server ({Constants.Processing.ResourceServerName})");
-				}
+				Console.WriteLine($"{nameof(CreateWorkerManagerServerAsync)} - Failed to create Worker Manager Server ({Constants.Processing.ResourceServerName}) as it already exists");
+				wasWorkerManagerServerCreated = true;

```

Comment:
Can we set this variable (wasWorkerManagerServerCreated) to true only once after the if/else statement, instead of doing it individually in each statement just like we did with the wasChoiceAddedToPool  variable above?

Reply:
Done